src/main/java/com/google/devtools/build/lib/remote/GrpcRemoteCache.java

```diff
@@ -289,12 +296,14 @@ public void onFailure(Throwable t) {
     Context ctx = Context.current();
     AtomicLong offset = new AtomicLong(0);
     ProgressiveBackoff progressiveBackoff = new ProgressiveBackoff(retrier::newBackoff);
-    return retrier.executeAsync(
-        () ->
-            ctx.call(
-                () ->
-                    requestRead(
-                        resourceName, offset, progressiveBackoff, digest, out, hashSupplier)));
+    return Futures.catchingAsync(
+        retrier.executeAsync(
+            () -> ctx.call(
+                () -> requestRead(resourceName, offset, progressiveBackoff, digest, out, hashSupplier)),
+            progressiveBackoff),
```

Comment:
why is this necessary?

Reply:
The previous implementation of the progressive read did not use the progressive backoff to determine the retry delay of successive requests. Put simply, we got progressive reads due to the offset, but there was no meaning to the progress from the perspective of the backoff used to proceed. A single download got max_retry_attempts to complete the download, and that's it - without this there was no effect to the progressiveBackoff being introduced.